code/modules/mob/living/simple_animal/bees.dm

```diff
@@ -98,45 +250,131 @@
+	update_icon()
+
 /mob/living/simple_animal/bee/proc/panic_attack(mob/damagesource)
 	for(var/mob/living/simple_animal/bee/B in range(src,3))
-		B.feral = 15
-		B.target = damagesource
-
-/mob/living/simple_animal/bee/wander_move(var/turf/dest)
-	var/goodmove=0
-	if(!my_hydrotray || my_hydrotray.loc != src.loc || my_hydrotray.dead || !my_hydrotray.seed)
-		// Wander the wastes
-		goodmove=1
-	else
-		// Restrict bee to area within distance of tray
-		var/turf/hiveturf = get_turf(my_hydrotray)
-		var/current_dist = get_dist(src,hiveturf)
-		var/new_dist = get_dist(dest,hiveturf)
-		// If we're beyond hive max range and we're not feral, we can only move towards or parallel to the hive.
-		if(current_dist > max_hive_dist && !feral)
-			if(new_dist <= current_dist)
-				goodmove=1
-		else
-			// Otherwise, we can move anywhere we like.
-			goodmove=1
-	if(goodmove)
-		Move(dest)
+		if (B.state == BEE_SWARM || calmed > 0)
+			return
+
+		//only their friends from the same apiary will answer their call. homeless bees will also help each others.
+		if (B.home == home)
+			B.state = BEE_OUT_FOR_ENEMIES
+			B.target = damagesource
+
 
-/mob/living/simple_animal/bee/proc/newTarget()
-	var/list/neabyMobs = list()
-	for(var/mob/living/G in view(src,7))
-		neabyMobs += G
-	target = pick(neabyMobs)
+////////////////////////////////LIFE////////////////////////////////////////
 
 /mob/living/simple_animal/bee/Life()
 	if(timestopped)
-		return 0 //under effects of time magick
+		return 0
 
 	..()
-	if(stat != DEAD) //If we're alive, see if we can be calmed down.
-		//smoke, water and steam calms us down
-		var/calming = 0
+
+	if (!bees || bees.len <= 0)
+		qdel(src)
+		return
+
+	if(stat != DEAD)
+		//SUFFERING FROM HIGH TOXICITY
+		if (((current_poison_damage - bees.len)/bees.len*100) > 50)
+			if (prob((((current_poison_damage - bees.len)/bees.len*100)-50)*2))
+				adjustBruteLoss(3)
+			if (bees.len <= 0)
+				return
+
+		//SPLITTING THE SWARM DEPENDING ON THEIR INTENT
+		if (updateState)
+			updateState = 0
+			var/list/swarmers = list()
+			var/list/home_goers = list()
+			var/list/pollinaters = list()
+			var/list/fighters = list()
+			var/turf/T = get_turf(src)
+
+			for (var/datum/bee/B in bees)
+				if (B.state == BEE_SWARM)
+					swarmers.Add(B)
+				if (B.state == BEE_HEADING_HOME)
+					home_goers.Add(B)
+				if (B.state == BEE_OUT_FOR_PLANTS)
+					pollinaters.Add(B)
+				if (B.state == BEE_OUT_FOR_ENEMIES)
+					fighters.Add(B)
+
+			if (swarmers.len > 0) // this intent comes from a queen, and thus overrides the intents of every other bee in the swarm
+				for (var/datum/bee/B in bees)
+					B.state = BEE_SWARM
+					B.home = destination
+				home = destination
+				state = BEE_SWARM
+
+			if (home_goers.len > 0)
+				if (home_goers.len == bees.len)
+					state = BEE_HEADING_HOME
+
+				else
+					var/mob/living/simple_animal/bee/B_mob = getFromPool(/mob/living/simple_animal/bee,T)
```

Comment:
Might be worth it modifying the /mob/living/simple_animal/bee/New() proc to accept a list of bee datums, and make it handle everything that is done below this line

Then there would be no need for the /mob/living/simple_animal/bee/adminspawned subtype, as if you wouldn't pass a list to New(), it could automatically create a single bee